### ISSUE TITLE
Clarification on 3D input shape order

### ISSUE THREAD

**Issue by Soongja** [ZFTurbo/timm_3d#5]
Hello, thank you for your amazing work! I’ve been trying out your timm-3d library.

In the README, the input shape is written as `(B, C, H, W, D)`, but I found this ordering a bit confusing.
PyTorch’s Conv3d uses `(B, C, D, H, W)`, and your volumentations library also follows `(D, H, W)`.
So I’m wondering, is `(H, W, D)` really the correct order?

If there is some internal code that permutes the dimensions, could you point it out?
And looking at these lines below, it seems like the order might be `(D, H, W)`(Since F.conv3d takes `(D, H, W)`):
https://github.com/ZFTurbo/timm_3d/blob/c79321d6b2766aa34c12d320e1703430fefbdb81/timm_3d/layers/cond_conv2d.py#L95-L103

Also, in my own experiments, I noticed that when I fed the data in `(D, H, W)` order as I initially assumed,
the validation score was actually higher.
I’m not sure if this is just a coincidence, so I’d really appreciate clarification on which order is the intended one.

Thanks a lot!

**Comment by doaaobeidat** [comment]
@Soongja did you get any answer?

**Comment by ZFTurbo** [comment]
Conv3D in most cases symmetric so you can think about any order you want. The difference can be only for non-symmetric padding/strides. Probably order here can be changed in this repo to be the same as in Conv3D.

**Comment by doaaobeidat** [comment]
@ZFTurbo 
This is my model
import timm_3d
import transformers
import copy
load_model = timm_3d.create_model(
   # 'densenet121',
   #'efficientnet_b0.ra_in1k',
    'tf_efficientnet_b0.ns_jft_in1k',
    #'tf_efficientnet_b0.in1k',
   
    #'mobilenetv2_140.ra_in1k',
    pretrained=True,
    num_classes=num_classes,
    global_pool='avg'
)
I used the input shape (B,C,D,H,W), so depth then height then width (96 slices, height 224, width 224) not 128x128x128
So my model is built based on pytorch and I used monai, my study (research) focusses on the effect of increasing/decreasing depth and image resolution, for that I am not sure if I can use depth*height*depth (96x224x244)
Note: I am asking in general, not for @Soongja post

These things differ with me, because if  I say in my research I used input of depth 96 and image 224x224 and the model will flip the order to take 96 as height not depth, this is a big problem. Can you help me please

**Comment by ZFTurbo** [comment]
You can use input as (B, C, 96, 224, 224) in this repository in case your dataset is of shape (96, 224, 224).

**Comment by doaaobeidat** [comment]
@ZFTurbo so is this true now

<img width="1056" height="593" alt="Image" src="https://github.com/user-attachments/assets/c3d484e6-bb06-472d-9d1a-4559b7b55e47" />

**Comment by ZFTurbo** [comment]
Yes, it should work fine like that.
